Lib/test/_test_multiprocessing.py

```diff
@@ -4724,6 +4724,40 @@ def test_level(self):
         root_logger.setLevel(root_level)
         logger.setLevel(level=LOG_LEVEL)
 
+    def assert_log_lines(self, expected_values, stream=None):
+        stream = stream or self.stream
+        actual_lines = stream.getvalue().splitlines()
+        self.assertEqual(len(actual_lines), len(expected_values))
+        for actual, expected in zip(actual_lines, expected_values):
+            self.assertEqual(actual, expected)
+        s = stream.read()
+        if s:
+            self.fail("Remaining output at end of log stream:\n" + s)
+
+    def test_filename(self):
+        logger = multiprocessing.get_logger()
+        default_level = logger.level
+        try:
+            logger.setLevel(util.DEBUG)
+            stream = io.StringIO()
+            handler = logging.StreamHandler(stream)
+            logging_format = '[%(levelname)s] [%(filename)s] %(message)s'
+            handler.setFormatter(logging.Formatter(logging_format))
+            logger.addHandler(handler)
+            logger.info('1')
+            util.info('2')
+            logger.debug('3')
+            filename = os.path.basename(__file__)
+            self.assert_log_lines([
```

Comment:
I think having a utility function might be a bit of an overkill here. Your check is very thorough but it might be too strict. I think something like

```python
output = stream.getvalue()
self.assertIn(f'[INFO] [{filename}] 1', output)
... 
```

should be good enough here. It's clear enough for people about what it's doing, and it's testing against the core issue this PR solves. Making sure the output matches *exactly* what's expected is not necessarily the best way to do it (especially if it makes the code harder to understand).